### ISSUE TITLE
Orphaned blocks containing pool payouts

### ISSUE THREAD

**Issue by zzzpotato** [s-nomp/s-nomp#99]
After the pool sends payouts, it does not verify if the payout was successfully sent and not contained within an orphaned block. When this happens, miners do not receive their payout yet it is recorded by the pool as sent.

S-nomp /api/payments
```
{
"time": 1543711226326,
"txid": "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595",
"shares": 281.62438554,
"paid": 356.8126093,
"miners": 3,
"blocks": [
412500
],
"amounts": {
"RZt7KmxmBZTqGV6UFNhuG8XBG7dg5afa7d": 350.05107276,
"Rk98sXCKeAnBEjP97yf2pMrPRi2zPTc7U1": 5.62171873,
"RhvHeso4AxaE1KLo1CNhiRV1RgTUEFK4Tm": 1.13981781
},
"balances": {
"Rn31SBJyN3X5e1tC5JvfUYL8Md2r3no9hU": 2.39008017
},
"work": {
"Rn31SBJyN3X5e1tC5JvfUYL8Md2r3no9hU": 56.375829949999996,
"RZt7KmxmBZTqGV6UFNhuG8XBG7dg5afa7d": 65.7613386,
"Rk98sXCKeAnBEjP97yf2pMrPRi2zPTc7U1": 132.60185315,
"RhvHeso4AxaE1KLo1CNhiRV1RgTUEFK4Tm": 26.88536384
}
},
```

Payment Daemon
```
safecoin-cli gettransaction "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595"
{
  "amount": -356.81260930,
  "fee": -0.00005000,
  "confirmations": -1,
  "txid": "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595",
```

**Comment by egyptianbman** [comment]
I've experienced this heavily with the SAFE chain over the last week. I'm working on a resolution to this on my pool right now. My idea is to scan the last 25-50 payments for `-1` confirmations. If found, I'll send out a new payout and replace the payout in the database with the new one.

I've found that the node sometimes reports the most recent payout as having `-1` confirmations. This is why I'm scanning the last 25-50 payments, instead of the the last 1-25.

I'm tempted to do two sets of a smaller window, say 25-30 and 75-80. This would allow for a "larger window" and not hit the daemon with 25 api calls on every payment loop. I'm still thinking through the final process but wanted to share an update.

**Comment by zzzpotato** [comment]
The resent payout will most likely need a recheck as well. I'm having issues with manual payouts being rejected.

**Comment by egyptianbman** [comment]
Yeah, the implementation I currently have in place would catch those since it posts the repayments with a key of the current time, putting them at the front of the list and would be rechecked as other payments are made.

**Comment by egyptianbman** [comment]
After putting some more thought into it, we don't need to search a range of payments (i.e. last 25 - 50) since this runs on every loop where one payment may have been made. What I have running now is a check for the most recent 25th, 50th and 75th payment for `-1` confirmations. I would expect the check for payout number 25 to catch all of them but I figure three checks isn't too bad.

I also have some code in there that's scanning all previous payments in the database and making payouts but I don't think I'll commit that since it needs manual intervention to run it properly.

**Comment by ashkanaslama** [comment]
@

**Comment by ashkanaslama** [comment]
{
"time": 1543711226326,
"txid": "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595",
"shares": 281.62438554,
"paid": 356.8126093,
"miners": 3,
"blocks": [
412500
],
"amounts": {
"RZt7KmxmBZTqGV6UFNhuG8XBG7dg5afa7d": 350.05107276,
"Rk98sXCKeAnBEjP97yf2pMrPRi2zPTc7U1": 5.62171873,
"RhvHeso4AxaE1KLo1CNhiRV1RgTUEFK4Tm": 1.13981781
},
"balances": {
"Rn31SBJyN3X5e1tC5JvfUYL8Md2r3no9hU": 2.39008017
},
"work": {
"Rn31SBJyN3X5e1tC5JvfUYL8Md2r3no9hU": 56.375829949999996,
"RZt7KmxmBZTqGV6UFNhuG8XBG7dg5afa7d": 65.7613386,
"Rk98sXCKeAnBEjP97yf2pMrPRi2zPTc7U1": 132.60185315,
"RhvHeso4AxaE1KLo1CNhiRV1RgTUEFK4Tm": 26.88536384
}
},
Payment Daemon

safecoin-cli gettransaction "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595"
{
  "amount": -356.81260930,
  "fee": -0.00005000,
  "confirmations": -1,
  "txid": "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595",

**Comment by ashkanaslama** [comment]
{
"time": 1543711226326,
"txid": "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595",
"shares": 281.62438554,
"paid": 356.8126093,
"miners": 3,
"blocks": [
412500
],
"amounts": {
"RZt7KmxmBZTqGV6UFNhuG8XBG7dg5afa7d": 350.05107276,
"Rk98sXCKeAnBEjP97yf2pMrPRi2zPTc7U1": 5.62171873,
"RhvHeso4AxaE1KLo1CNhiRV1RgTUEFK4Tm": 1.13981781
},
"balances": {
"Rn31SBJyN3X5e1tC5JvfUYL8Md2r3no9hU": 2.39008017
},
"work": {
"Rn31SBJyN3X5e1tC5JvfUYL8Md2r3no9hU": 56.375829949999996,
"RZt7KmxmBZTqGV6UFNhuG8XBG7dg5afa7d": 65.7613386,
"Rk98sXCKeAnBEjP97yf2pMrPRi2zPTc7U1": 132.60185315,
"RhvHeso4AxaE1KLo1CNhiRV1RgTUEFK4Tm": 26.88536384
}
},
Payment Daemon

safecoin-cli gettransaction "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595"
{
  "amount": -356.81260930,
  "fee": -0.00005000,
  "confirmations": -1,
  "txid": "32f15f26a6056652e0b66a4b46f696a6269856314035c9c38b6d9799cb6c4595",
